### PR TITLE
hs.fs tests and refactoring

### PR DESCRIPTION
**Pull request by heptal**
hs.fs coverage to 96%
hs.fs.volume coverage to 97%
-Weverything -Werror
Removed win32 code and other such unused `#define`s
Reindented

### CODE REVIEW

File: extensions/fs/test_fs.lua [hunk 1]
```diff
@@ -0,0 +1,331 @@
+-- hs.fs tests
+
+homeDir = os.getenv("HOME").."/"
```
Comment (cmsj) [inline extensions/fs/test_fs.lua]: I'm a little nervous about depending on the environment for a path, and then appending "/" to it, it has all the makings of a "rm -r /" bug.

How about we just hard code to using `/tmp/hs_fs_test_dir/`?

**Comment by cmsj** [comment]
Looking good, just had one comment :)

**Comment by heptal** [comment]
Fixed

**Comment by cmsj** [comment]
Thanks :)

(I tweaked it ever-so-slightly to just use `/tmp/` - I'm not sure if we're guaranteed that `/private/tmp`, which is after-all the same actual folder - will remain where it is)

**Pull request closed without merging**